.github/workflows/event-merge-queue-resubmit.yml

```diff
@@ -0,0 +1,127 @@
+          # Get the PR's actual head SHA (the last commit on the PR branch)
+          PR_HEAD_SHA=$(gh api "/repos/$REPO/pulls/$PR_NUM" --jq '.head.sha')
+          echo "PR's actual HEAD SHA: $PR_HEAD_SHA"
+
+          # Get the list of previous workflow runs for this merge queue workflow
+          # Look for runs from the same PR that have failed/cancelled
+          echo "Searching for previous merge_group runs for PR #$PR_NUM..."
+
+          ALL_RUNS=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "/repos/$REPO/actions/runs?event=merge_group&per_page=100" \
+            --jq ".workflow_runs[] | select(.id != $CURRENT_RUN_ID) | {id: .id, head_sha: .head_sha, conclusion: .conclusion, head_ref: .head_branch}" | jq -s '.')
+
+          echo "Recent merge_group runs:"
+          echo "$ALL_RUNS" | jq -c '.[]'
+
+          # Look for previous runs from the same PR (by matching PR number in head_ref)
+          # Use regex anchors to ensure exact PR number match (e.g., pr-12- won't match pr-123-)
+          PREVIOUS_RUN_DATA=$(echo "$ALL_RUNS" | jq -c ".[] | select(.head_ref | test(\"(^|/)pr-${PR_NUM}-\")) | select(.conclusion == \"failure\" or .conclusion == \"cancelled\")" | head -1)
+
+          if [[ -z "$PREVIOUS_RUN_DATA" ]]; then
+            echo "No previous failed/cancelled runs found for PR #$PR_NUM"
+            echo "resubmit=false" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+
+          PREVIOUS_RUN_ID=$(echo "$PREVIOUS_RUN_DATA" | jq -r '.id // empty')
+          echo "Found previous failed/cancelled run ID: $PREVIOUS_RUN_ID"
+
+          # Now check if the PR's head SHA has changed since that previous run
+          # Get the PR's head SHA at the time of the previous run by fetching the run details
+          PREVIOUS_RUN_DETAILS=$(gh api \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "/repos/$REPO/actions/runs/$PREVIOUS_RUN_ID")
+
+          # Extract the PR head SHA from the previous run's jobs
+          # The merge queue creates a merge commit, but we need to check the actual PR commits
+          # We'll get the commit list from the previous run and find the PR's head commit
+          PREVIOUS_RUN_CREATED_AT=$(echo "$PREVIOUS_RUN_DETAILS" | jq -r '.created_at')
+          echo "Previous run was created at: $PREVIOUS_RUN_CREATED_AT"
+
+          # Get the PR's commit history and find what the head SHA was at the time of the previous run
+          # We'll check if the current PR head SHA existed before the previous run
+          COMMIT_DATE=$(gh api "/repos/$REPO/commits/$PR_HEAD_SHA" --jq '.commit.committer.date')
+          echo "Current PR HEAD commit date: $COMMIT_DATE"
+
+          # Compare dates: if the current PR head commit is newer than the previous run, there were new commits
+          if [[ "$COMMIT_DATE" > "$PREVIOUS_RUN_CREATED_AT" ]]; then
```

Comment:
What's the granularity here? No just the day, I hope

Reply:
No, it's a full time stamp in daytime format